Documentation/Makefile

```diff
@@ -44,6 +44,7 @@ endif
 # documentation to be generated correctly.
 cilium-build:
 ifndef SKIP_BUILD
+	$(MAKE) builder-image
```

Comment:
Why not make `builder-image` a dependency of `cilium-build:` (instead of recursively calling `make`)?

Reply:
I think I went to do that, but noticed the `ifndef` here and figured that there's no need to trigger the `builder-image` dependency if SKIP_BUILD is set. Not quite sure why but there's another _different_ env var protecting that target in case developers don't want to trigger that build, but :shrug: tomato tomato.